internal/worker/units3caller/manifold.go

```diff
@@ -107,3 +106,14 @@ func (c *httpClient) Do(req *http.Request) (*http.Response, error) {
 	err := c.client.Do(req.Context(), req, &res)
 	return res, err
 }
+
+// ensureHTTPS takes a URI and ensures that it is a HTTPS URL.
+func ensureHTTPS(address string) string {
```

Comment:
Seem to recall we do this for LXD. Worth making a utility function in core somewhere?

Reply:
I'll do that in the next PR